go-controller/pkg/ovn/gateway_init.go

```diff
@@ -614,6 +623,10 @@ func syncPolicyBasedRoutes(nodeName, mgmtPortIP, match, priority, nexthops, comp
 					return fmt.Errorf("failed to delete policy route '%s' for host %q on %s "+
 						"error: %v", match, nodeName, types.OVNClusterRouter, err)
 				}
+				// We should only ever find one LRP for a given node and priority,
+				// so the following should be a noop. But you never know...
```

Comment:
I'm not sure what this comment means. You are looping through the policies to find the 1 that belongs to the node. The presumption here is that there is only one LRP for a node and priority, that's true, but that has nothing to do with the next line of setting the found value to false. That only pertains to breaking the loop because you found an entry for a node with an invalid mgmtport IP and deleted it. The comment is confusing because it occurs before the found=false line, which has nothing to do with the comment.

found should be false, because you know you need to create the right one, after you deleted the wrong one. I would suggest moving the comment to the doctext of the function and saying "This function assumes one LRP for a given node and priority exists", and getting rid of this inline comment.

Reply:
Done

Reply:
FWIW: that comment referenced the fact that setting `found = false` should be useless, unless something bad has happened (where we have multiple LRPs for the same node and priority). Since there should only be one LRP for a given priority and node: the `break` statement is the only thing really needed. If we would reach the `break`: `found` should remain unchanged and thus as initialized: `false`.